indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -206,16 +208,21 @@ def _send_emails(self, form):
             template = get_template_module('events/registration/emails/custom_email.html',
                                            email_subject=form.subject.data, email_body=email_body)
             bcc = [session.user.email] if form.copy_for_sender.data else []
+            attachments = get_ticket_attachment(registration) if form.attach_ticket.data else None
             email = make_email(to_list=registration.email, cc_list=form.cc_addresses.data, bcc_list=bcc,
-                               from_address=form.from_address.data, template=template, html=True)
+                               from_address=form.from_address.data, template=template, html=True,
+                               attachments=attachments)
             send_email(email, self.event, 'Registration')
 
     def _process(self):
         tpl = get_template_module('events/registration/emails/custom_email_default.html')
         default_body = tpl.get_html_body()
         registration_ids = request.form.getlist('registration_id')
         form = EmailRegistrantsForm(body=default_body, regform=self.regform, registration_id=registration_ids,
-                                    recipients=[x.email for x in self.registrations])
+                                    recipients=[x.email for x in self.registrations], )
+        if self.regform.tickets_enabled:
+            setattr(EmailRegistrantsForm, 'attach_ticket', BooleanField(_('Attach ticket'), widget=SwitchWidget(),
```

Comment:
⚠️ This modifies the form class and will affect ALL EVENTS (for requests handled by the same process).

You need to subclass the form class on the fly:

    form_class = type(b'_EmailRegistrantsForm', (EmailRegistrantsForm,), {})

and then set `form_class.attach_ticket = ...` (no need to ever use setattr with a non-dynamic second argument) and use `form = form_class(...)`

---

However, You could just add the field to the form class like any other field and `del form.attach_ticket` if tickets are *not* enabled. That's much easier and thus probably the better solution here.